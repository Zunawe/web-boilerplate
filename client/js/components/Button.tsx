import React, { FC, ReactNode, useCallback } from 'react'

interface ButtonProps {
  children: ReactNode
  onClick?: () => void
}

const Button: FC<ButtonProps> = ({ onClick, children }) => {
  const handleClick = useCallback((event: React.MouseEvent<HTMLButtonElement, MouseEvent>) => {
    event.preventDefault()
    onClick?.()
  }, [onClick])

  return (
    <button onClick={handleClick}>{children}</button>
  )
}

export default Button
