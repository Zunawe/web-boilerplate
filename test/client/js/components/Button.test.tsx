import React from 'react'
import { render, fireEvent, screen } from '@testing-library/react'

import { Button } from '../../../../client/js/components/Button'

describe('Button', () => {
  it('should render a default button with text', () => {
    const { container } = render(<Button>Button</Button>)
    expect(container.firstChild).toMatchSnapshot()
  })

  it('should run the click handler', () => {
    const handleClick = jest.fn()
    render(<Button onClick={handleClick}>Button</Button>)
    fireEvent.click(screen.getByText('Button'))
    expect(handleClick.mock.calls.length).toBe(1)
  })
})
