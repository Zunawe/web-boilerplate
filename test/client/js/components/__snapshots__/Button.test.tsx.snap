// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render a default button with text 1`] = `
<button>
  Button
</button>
`;
